src/transformers/modeling_utils.py

```diff
@@ -2820,6 +2820,13 @@ def set_attn_implementation(self, attn_implementation: Union[str, dict]):
                 and isinstance(submodule, PreTrainedModel)
                 and submodule.config.__class__ != self.config.__class__
             ):
+                if not submodule._can_set_attn_implementation:
+                    logger.warning(
```

Comment:
let's only warn once then

Reply:
Okay I was thinking like from_pretrained should only warn once, but we can do it with `is_init_check`.